toolchain/papiTest.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM.
+# Author: Umesh S <umess1@linux.vnet.ibm.com>
+
+
+import os
+from avocado.utils import process
+from avocado import Test
+from avocado import main
+from avocado.utils import git, build
+from avocado.utils.software_manager import SoftwareManager
+
+
+class papitest(Test):
+
+    '''
+    This testcase make use of testsuite provided by the
+    source package, source files are downloaded and compiled
+    '''
+
+    def setUp(self):
+
+        softm = SoftwareManager()
+
+        for package in ['gcc', 'make']:
+            if not softm.check_installed(package) and not softm.install(package):
+                self.cancel("%s is needed for the test to be run" % package)
+        test_type = self.params.get('type', default='distro')
+
+        if test_type == 'upstream':
+            git.get_repo('https://github.com/arm-hpc/papi.git',
+                         destination_dir=self.teststmpdir)
+            self.path = os.path.join(self.teststmpdir, 'src')
+        elif test_type == 'distro':
+            sourcedir = os.path.join(self.teststmpdir, 'papi-distro')
+            if not os.path.exists(sourcedir):
+                os.makedirs(sourcedir)
+            self.path = softm.get_source("papi", sourcedir)
+
+        os.chdir(self.path)
+        process.run('./configure', shell=True)
+        build.make(self.path)
+
+    def test(self):
+
+        #Runs the tests
+
+        result = process.run('./run_tests.sh', shell=True)
+
+        #Display the failed tests
+
+        errors = 0
+        errlist = []
+        list.append(result.stdout.splitlines())
+        for inlist in errlist:
+            for text in inlist:
+                if 'FAILED' in text or 'WARNING' in text:
+                    self.log.info(text)
+                    errors += 1
+
+        self.log.info("Total number of failed cases is %s ", errors)
```

Comment:
can we have self.fail for this test

Reply:
I didn't use self.fail because am also considering tests 'PASSED WITH WARNING' along with the 'FAILED' cases. I'll miss out those cases on using self.fail .

Reply:
sure than i think we can have two section one is `PASSED WITH WARNING` as using `self.log.warn()` and failure as self.fail what you think

Reply:
I have made the necessary changes..